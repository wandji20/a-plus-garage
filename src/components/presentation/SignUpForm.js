import React, { useState } from 'react';
import PropTypes from 'prop-types';
// import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { Link, Redirect } from 'react-router-dom';
import FormError from './FormError';
import signUpUserAction from '../../redux/actions/signUpAction';

const SignUpForm = (props) => {
  const { handleSignUpUser, response } = props;
  console.log(response);
  const [name, setName] = useState('');
  const [userID, setUserID] = useState('');

  const handleNameChange = (e) => {
    setName(e.target.value);
  };

  const handleUserIDChange = (e) => {
    setUserID(e.target.value);
  };

  const handleFormSubmit = (e) => {
    const data = { name, userID };
    e.preventDefault();
    setName('');
    setUserID('');
    handleSignUpUser(data);
  };

  return (
    <div className="container w-75 d-flex flex-column justify-content-center align-items-left">
      {
      response.success
        ? <Redirect to="/" />
        : (
          <>
            <form
              className=" d-flex flex-column align-items-start col-sm-8 col-md-6"
              onSubmit={handleFormSubmit}
            >
              <div className="form-group my-3">
                <label htmlFor="name">
                  User Name
                  <input
                    type="text"
                    id="name"
                    value={name}
                    className="form-control"
                    placeholder="...name"
                    onChange={handleNameChange}
                  />
                </label>
                {
              (!response.success && response.errors && response.errors.name)
                && <FormError column="User name" errors={response.errors.name} />
            }

              </div>
              <div className="form-group my-3">
                <label htmlFor="formGroupExampleInput2">
                  UserID
                  <input
                    type="text"
                    value={userID}
                    className="form-control"
                    placeholder="@username"
                    onChange={handleUserIDChange}
                  />
                </label>
                {
              (!response.success && response.errors && response.errors.userID) && <FormError column="UserID" errors={response.errors.userID} />
            }
              </div>
              <button type="submit" className="btn btn-primary mb-2">Sign Up</button>
            </form>
            <p className="">
              <Link to="/log_in" style={{ color: 'black', textDecoration: 'none' }}>
                Already have an account ? Login
              </Link>
            </p>
          </>
        )
    }
    </div>
  );
};

const mapStateToProps = (state) => ({
  response: state.userReducer.response,
});

const mapDispatchToProps = (dispatch) => ({
  handleSignUpUser: (data) => {
    dispatch(signUpUserAction(data));
  },
});

SignUpForm.propTypes = {
  // history: PropTypes.objectOf(PropTypes.any).isRequired,
  response: PropTypes.objectOf(PropTypes.any).isRequired,
  handleSignUpUser: PropTypes.func,
};

SignUpForm.defaultProps = {
  handleSignUpUser: () => {},
};

// export default connect(mapStateToProps, mapDispatchToProps)(withRouter(SignUpForm));
export default connect(mapStateToProps, mapDispatchToProps)(SignUpForm);
