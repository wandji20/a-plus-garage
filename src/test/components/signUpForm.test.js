import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import SignUpForm from '../../components/presentation/SignUpForm';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <SignUpForm />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
