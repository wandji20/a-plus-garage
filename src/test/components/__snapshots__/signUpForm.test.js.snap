// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container w-75 d-flex flex-column justify-content-center align-items-left"
>
  <p
    className="row justify-content-center text-danger"
  />
  <form
    className=" d-flex flex-column align-items-start col-sm-8 col-md-6"
    onSubmit={[Function]}
  >
    <div
      className="form-group my-3"
    >
      <label
        htmlFor="name"
      >
        Name
        <input
          className="form-control"
          id="name"
          onChange={[Function]}
          placeholder="...name"
          required={true}
          type="text"
          value=""
        />
      </label>
    </div>
    <div
      className="form-group my-3"
    >
      <label
        htmlFor="formGroupExampleInput2"
      >
        User Name
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="@username"
          required={true}
          type="text"
          value=""
        />
      </label>
    </div>
    <div
      className="form-group my-3"
    >
      <label
        htmlFor="formGroupExampleInput2"
      >
        Password
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="password"
          required={true}
          type="password"
          value=""
        />
      </label>
    </div>
    <div
      className="form-group my-3"
    >
      <label
        htmlFor="formGroupExampleInput2"
      >
        Confirm Password
        <input
          className="form-control"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </label>
    </div>
    <button
      className="btn btn-primary mb-2"
      type="submit"
    >
      Sign Up
    </button>
  </form>
  <p
    className=""
  >
    <span
      className="inline-block"
    >
      Already have an account ? 
    </span>
    <span
      className="inline-block"
    >
      <a
        href="/log_in"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        log in
      </a>
    </span>
  </p>
</div>
`;
